docs/pages/debugging/creating-devtools-plugins.mdx

```diff
@@ -0,0 +1,109 @@
+---
+title: Creating Dev Tools Plugins
+description: Learn how to make your own dev tools plugin to enhance your development experience.
+---
+
+import { Terminal } from '~/ui/components/Snippet';
+
+> Dev Tools plugins are compatible with SDK 50 and above. The process for creating a plugin is still in beta and may change in the future.
+
+You can make your own dev tools plugin, whether that's for inspecting aspects of a common framework or library, or something specific to your custom code.
+
+A plugin consists of an Expo app for displaying the dev tool's web user interface, an **expo-module.config.json** file to help the Expo CLI detect the plugin, and then calls to `expo/devtools` API to help your app communicate back and forth with the dev tool's web UI. Plugins can be included inside your app's monorepo or distributed as a separate package on NPM.
+
+## Creating a plugin
+
+### Create an Expo app
+
+The web interface for your dev tool will be an Expo app. Create a new blank Expo app:
```

Comment:
from the design philosophy, i don't want to have a strong opinion for the web ui. we could use any web libs or frameworks, not just expo apps.

for proof-of-concept, i created an example of using vanilla-js as webui: https://github.com/expo/dev-plugins/tree/main/packages/vanilla-log-viewer